src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -24,7 +24,7 @@ public static bool IsInvalid(this IOperation operation, Compilation compilation,
 
             // if wrong compilation is given, GetSemanticModel will throw due to tree not belong to the given compilation.
             var model = compilation.GetSemanticModel(operation.Syntax.SyntaxTree);
-            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.Severity == DiagnosticSeverity.Error);
+            return model.GetDiagnostics(operation.Syntax.Span, cancellationToken).Any(d => d.DefaultSeverity == DiagnosticSeverity.Error);
```

Comment:
I don't really have a strong preference on the name, but I do think this property should be used to indicate whether the operation is based on code that is correct (correct as free of compile errors, not error-level diagnostics).